### ISSUE TITLE
Could not read file exception with 1.8.10

### ISSUE THREAD

**Issue by FrancoisBlavoet** [Kotlin/dokka#2956]
**Describe the bug**
I am migrating my android library project to kotlin 1.8.10, but I started getting this exception: 
```
Could not read file: /nvme/buildkite/builds/8192mb-4096cpu-android_603ae6562927_fc6092f/instacart/instacart-design-system-android/legacy-core/build/intermediates/compile_library_classes_jar/release/classes.jar!/com/instacart/design/itemcard/ItemCard.class; size in bytes: 3691; file type: CLASS
--
  | java.lang.UnsupportedOperationException: PermittedSubclasses requires ASM9

```

**Expected behaviour**

Not crashing would be very nice and polite

**To Reproduce**
It seems to occur when a module refers to classes from another module

**Dokka configuration**

```Kotlin 1.8.10

```

**Installation**
- Operating system: macOS/Linux
- Build tool: Gradle 8.0.2, happens on both CI and locally
- Dokka version: 1.8.10 (I have another error if I try to only downgrade dokka to 1.7.20)

**Additional context**

**Are you willing to provide a PR?**
Potentially yes, I might need some guidance to pinpoint the source thoug

**Comment by FrancoisBlavoet** [comment]
could this be a duplicate of https://github.com/Kotlin/dokka/issues/2877 ?

**Comment by IgnatBeresnev** [comment]
Can you please try running it with a different version of Java? Ideally, we'd need an isolated reproducer project as well, or at least the full stacktrace - at the moment it's unclear if it's Dokka's fault (we use a few dependencies for code analysis), and where (how) exactly it happens

**Comment by FrancoisBlavoet** [comment]
@IgnatBeresnev Thanks for the reply!
I have been waiting for dokka 1.8.20 in the hope that it might fix this (it apparently addresses a relatively similar pb) 

In the meantime, I have just tried with gradle 8.1:


same result unfortunately


> Task :legacy-bridge:javaDocReleaseGeneration
  | Initializing plugins
  | Dokka is performing: documentation for :legacy-bridge
  | Validity check
  | Creating documentation models
  |  
  | WARN: Could not read file: /nvme/buildkite/builds/8192mb-4096cpu-android_1becc3d97419_18b51ed/instacart/instacart-design-system-android/legacy-core/build/intermediates/compile_library_classes_jar/release/classes.jar!/com/instacart/design/itemcard/ItemCard.class; size in bytes: 3805; file type: CLASS
  | java.lang.UnsupportedOperationException: PermittedSubclasses requires ASM9
  | at org.jetbrains.org.objectweb.asm.ClassVisitor.visitPermittedSubclass(ClassVisitor.java:266)
  | at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:684)
  | at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:402)
  | at org.jetbrains.kotlin.load.kotlin.FileBasedKotlinClass.create(FileBasedKotlinClass.java:96)
  | at org.jetbrains.kotlin.load.kotlin.VirtualFileKotlinClass$Factory$create$1.invoke(VirtualFileKotlinClass.kt:67)
  | at org.jetbrains.kotlin.load.kotlin.VirtualFileKotlinClass$Factory$create$1.invoke(VirtualFileKotlinClass.kt:61)
  | at org.jetbrains.kotlin.util.PerformanceCounter.time(PerformanceCounter.kt:101)
  | at org.jetbrains.kotlin.load.kotlin.VirtualFileKotlinClass$Factory.create(VirtualFileKotlinClass.kt:61)
  | at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion$getKotlinBinaryClassOrClassFileContent$aClass$1.compute(KotlinBinaryClassCache.kt:75)
  | at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion$getKotlinBinaryClassOrClassFileContent$aClass$1.compute(KotlinBinaryClassCache.kt:73)
  | at com.intellij.mock.MockApplication.runReadAction(MockApplication.java:178)
  | at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion.getKotlinBinaryClassOrClassFileContent(KotlinBinaryClassCache.kt:73)
  | at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion.getKotlinBinaryClassOrClassFileContent$default(KotlinBinaryClassCache.kt:59)
  | at org.jetbrains.kotlin.load.kotlin.VirtualFileFinder.findKotlinClassOrContent(VirtualFileFinder.kt:34)
  | at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope$classes$1.invoke(LazyJavaPackageScope.kt:62)
  | at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope$classes$1.invoke(LazyJavaPackageScope.kt:54)
  | at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
  | at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope.findClassifier(LazyJavaPackageScope.kt:142)
  | at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope.getContributedClassifier(LazyJavaPackageScope.kt:132)
  | at org.jetbrains.kotlin.load.java.lazy.descriptors.JvmPackageScope.getContributedClassifier(JvmPackageScope.kt:55)
  | at org.jetbrains.kotlin.resolve.scopes.ChainedMemberScope.getContributedClassifier(ChainedMemberScope.kt:35)
  | at org.jetbrains.kotlin.resolve.scopes.AbstractScopeAdapter.getContributedClassifier(AbstractScopeAdapter.kt:44)
  | at org.jetbrains.kotlin.resolve.LazyExplicitImportScope.getContributedClassifier(LazyExplicitImportScope.kt:43)
  | at org.jetbrains.kotlin.resolve.lazy.LazyImportScope$getClassifier$1.invoke(LazyImportScope.kt:256)
  | at org.jetbrains.kotlin.resolve.lazy.LazyImportScope$getClassifier$1.invoke(LazyImportScope.kt:251)
  | at org.jetbrains.kotlin.storage.LockBasedStorageManager.compute(LockBasedStorageManager.java:290)
  | at org.jetbrains.kotlin.resolve.lazy.LazyImportScope.getClassifier(LazyImportScope.kt:251)
  | at org.jetbrains.kotlin.resolve.lazy.LazyImportScope.getContributedClassifier(LazyImportScope.kt:247)
  | at org.jetbrains.kotlin.resolve.scopes.ResolutionScope$DefaultImpls.getContributedClassifierIncludeDeprecated(ResolutionScope.kt:40)
  | at org.jetbrains.kotlin.resolve.scopes.HierarchicalScope$DefaultImpls.getContributedClassifierIncludeDeprecated(Scopes.kt:27)
  | at org.jetbrains.kotlin.resolve.scopes.ImportingScope$DefaultImpls.getContributedClassifierIncludeDeprecated(Scopes.kt:110)
  | at org.jetbrains.kotlin.resolve.lazy.LazyImportScope.getContributedClassifierIncludeDeprecated(LazyImportScope.kt:220)
  | at org.jetbrains.kotlin.resolve.scopes.utils.ScopeUtilsKt.findFirstClassifierWithDeprecationStatus(ScopeUtils.kt:117)
  | at org.jetbrains.kotlin.resolve.QualifiedExpressionResolver.findClassifierAndReportDeprecationIfNeeded(QualifiedExpressionResolver.kt:82)
  | at org.jetbrains.kotlin.resolve.QualifiedExpressionResolver.resolveDescriptorForType(QualifiedExpressionResolver.kt:106)
  | at org.jetbrains.kotlin.resolve.TypeResolver.resolveDescriptorForType(TypeResolver.kt:931)
  | at org.jetbrains.kotlin.resolve.TypeResolver$resolveTypeElement$1.visitUserType(TypeResolver.kt:246)
  | at org.jetbrains.kotlin.psi.KtVisitorVoid.visitUserType(KtVisitorVoid.java:929)
  | at org.jetbrains.kotlin.psi.KtVisitorVoid.visitUserType(KtVisitorVoid.java:21)
  | at org.jetbrains.kotlin.psi.KtUserType.accept(KtUserType.java:42)
  | at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:60)
  | at org.jetbrains.kotlin.resolve.TypeResolver.resolveTypeElement(TypeResolver.kt:244)
  | at org.jetbrains.kotlin.resolve.TypeResolver.resolvePossiblyBareType(TypeResolver.kt:123)
  | at org.jetbrains.kotlin.resolve.TypeResolver.resolveType(TypeResolver.kt:113)
  | at org.jetbrains.kotlin.resolve.TypeResolver.resolveType(TypeResolver.kt:84)
  | at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.resolveValueParameters(FunctionDescriptorResolver.kt:421)
  | at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.createValueParameterDescriptors(FunctionDescriptorResolver.kt:298)
  | at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.initializeFunctionDescriptorAndExplicitReturnType(FunctionDescriptorResolver.kt:191)
  | at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.resolveFunctionDescriptor(FunctionDescriptorResolver.kt:124)
  | at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.resolveFunctionDescriptor(FunctionDescriptorResolver.kt:91)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.getDeclaredFunctions(AbstractLazyMemberScope.kt:115)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.access$getDeclaredFunctions(AbstractLazyMemberScope.kt:36)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$declaredFunctionDescriptors$1.invoke(AbstractLazyMemberScope.kt:56)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$declaredFunctionDescriptors$1.invoke(AbstractLazyMemberScope.kt:56)
  | at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
  | at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunctionToNotNull.invoke(LockBasedStorageManager.java:651)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.doGetFunctions(AbstractLazyMemberScope.kt:96)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.access$doGetFunctions(AbstractLazyMemberScope.kt:36)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$functionDescriptors$1.invoke(AbstractLazyMemberScope.kt:49)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$functionDescriptors$1.invoke(AbstractLazyMemberScope.kt:49)
  | at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
  | at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunctionToNotNull.invoke(LockBasedStorageManager.java:651)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.getContributedFunctions(AbstractLazyMemberScope.kt:92)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.computeDescriptorsFromDeclaredElements(AbstractLazyMemberScope.kt:227)
  | at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyPackageMemberScope.getContributedDescriptors(LazyPackageMemberScope.kt:44)
  | at org.jetbrains.kotlin.resolve.scopes.ResolutionScope$DefaultImpls.getContributedDescriptors$default(ResolutionScope.kt:50)
  | at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor.getDescriptorsWithKind(DefaultDescriptorToDocumentableTranslator.kt:687)
  | at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor$visitPackageFragmentDescriptor$2.invokeSuspend(DefaultDescriptorToDocumentableTranslator.kt:131)
  | at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor$visitPackageFragmentDescriptor$2.invoke(DefaultDescriptorToDocumentableTranslator.kt)
  | at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
  | at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:194)
  | at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor.visitPackageFragmentDescriptor(DefaultDescriptorToDocumentableTranslator.kt:130)
  | at org.jetbrains.dokka.base.translators.descriptors.DefaultDescriptorToDocumentableTranslator$special$inlined$parallelMap$1$1.invokeSuspend(parallelCollectionOperations.kt:19)
  | at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
  | at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
  | at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
  | at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
  | at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
  | at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
  |  
  | > Task :legacy-bridge:javaDocReleaseGeneration FAILED
  |  
  |  
  | FAILURE: Build failed with an exception.
  |  
  | * What went wrong:
  | Execution failed for task ':legacy-bridge:javaDocReleaseGeneration'.
  | > A failure occurred while executing com.android.build.gradle.tasks.JavaDocGenerationTask$DokkaWorkAction
  | > java.lang.reflect.InvocationTargetException (no error message)
 

<br class="Apple-interchange-newline">

**Comment by FrancoisBlavoet** [comment]
I suspect that this is tied to the kotlin version, dokka works fine on the main branch of my project with kotlin 1.7.10

**Comment by FrancoisBlavoet** [comment]
I tried to toggle more features on/off
it turns out that I don't have this problem if I set the project's java language level to 11 (instead of 18 previously)

**Comment by trevjonez** [comment]
> Among other things because JDK 17 will soon be necessary to run the android gradle plugin

AGP 8.0.0 is stable now so we are now there.

**Comment by IgnatBeresnev** [comment]
Thanks for researching it!

```
Execution failed for task ':legacy-bridge:javaDocReleaseGeneration'.
> A failure occurred while executing com.android.build.gradle.tasks.JavaDocGenerationTask$DokkaWorkAction
   > java.lang.reflect.InvocationTargetException (no error message)
```

This is interesting. Are you using a task called `javaDocReleaseGeneration`? If yes, then you are not really using Dokka (at least not directly), as Dokka's tasks are `dokkaHtml` and `dokkaHtmlMultiModule`.

AGP 7.1 did start shipping a very old version Dokka, which led to a bunch of compatibility problems for existing Dokka users (see #2472), so here there might be a similar problem here.

Judging by the stacktrace, it looks like the version of the kotlin compiler used to generate the docs does not support Java 18. If AGP ships a very outdated version of Dokka, no wonder!

___

I've just tried running Dokka with Java 19, and it works just fine:

```kotlin
plugins {
    kotlin("jvm") version "1.8.10"
    id("org.jetbrains.dokka") version "1.8.10"
}

tasks.named("compileKotlin", org.jetbrains.kotlin.gradle.tasks.KotlinJvmCompile::class.java) {
    compilerOptions {
        jvmTarget.set(JvmTarget.JVM_19)
    }
}
```

Log:

```
[ignat@workstation dokka-debug-kts]$ java -version

openjdk version "19.0.2" 2023-01-17
OpenJDK Runtime Environment Temurin-19.0.2+7 (build 19.0.2+7)
OpenJDK 64-Bit Server VM Temurin-19.0.2+7 (build 19.0.2+7, mixed mode, sharing)

[ignat@workstation dokka-debug-kts]$ ./gradlew dokkaHtml

> Task :dokkaHtml
Initializing plugins
Dokka is performing: documentation for dokka-debug-kts
Validity check
Creating documentation models
Transforming documentation model before merging
Merging documentation models
Transforming documentation model after merging
Creating pages
Transforming pages
Rendering
Running post-actions


BUILD SUCCESSFUL in 10s
1 actionable task: 1 executed
```

___

So it might be an upstream bug of AGP.

Can you share which task you're using just to make sure?

**Comment by FrancoisBlavoet** [comment]
interesting, so we have had more time to dig into this as we need to find a solution to adopt compose 1.5.
We made some updates to our project -> Compile SDK 34 + Java 18 + Kotlin 1.8.22 + paparazzi 1.3.0 + AGP 8.1 + vanniktech publish plugin 0.25.3 

This has allowed us to dig deeper into the doc generation as we still see this issue. 

we are indeed using `javaDocReleaseGeneration`  (via the vanniktech publish plugin) 

our `solution' has been to disable this task for the one specific module where this happens 
This is an old part of our library we don't really care about, so this is an acceptable compromise to unblock the updates. The weird thing is that there isn't anything that remarkable about this module. it depends on other modules, but so do other parts of our library, without causing this issue. 

So I have tried to use `dokkaHtmlMultiModule` instead, if only to help understand what's happening. 
This works! My suspicion is that we are hitting a variation of  https://github.com/Kotlin/dokka/issues/1833

We will investigate further and decided whether to contribute a fix at the vanniktech level or to migrate to our own internal publish plugin (we already have one, still incubating). Hard to know for sure without digging more into its codebase, but I think that the core issue is due to how vanniktech declares dependencies between modules.

**Comment by nate-eisner** [comment]
Just updated a project to AGP 8.1.1 and Kotlin 1.9.10 and getting the same error when I try to publish to maven local. We aren't using anything fancy, just `maven-publish` plugin,

```
> Task :components:javaDocReleaseGeneration
Initializing plugins
Dokka is performing: documentation for :components
Validity check
Creating documentation models

WARN: Could not read file: compile_library_classes_jar/release/classes.jar!/com/x/x/components/ListItem.class; size in bytes: 12858; file type: CLASS
java.lang.UnsupportedOperationException: PermittedSubclasses requires ASM9
	at org.jetbrains.org.objectweb.asm.ClassVisitor.visitPermittedSubclass(ClassVisitor.java:266)
	at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:684)
	at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:402)
	at org.jetbrains.kotlin.load.kotlin.FileBasedKotlinClass.create(FileBasedKotlinClass.java:96)
	at org.jetbrains.kotlin.load.kotlin.VirtualFileKotlinClass$Factory$create$1.invoke(VirtualFileKotlinClass.kt:67)
	at org.jetbrains.kotlin.load.kotlin.VirtualFileKotlinClass$Factory$create$1.invoke(VirtualFileKotlinClass.kt:61)
	at org.jetbrains.kotlin.util.PerformanceCounter.time(PerformanceCounter.kt:101)
	at org.jetbrains.kotlin.load.kotlin.VirtualFileKotlinClass$Factory.create(VirtualFileKotlinClass.kt:61)
	at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion$getKotlinBinaryClassOrClassFileContent$aClass$1.compute(KotlinBinaryClassCache.kt:75)
	at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion$getKotlinBinaryClassOrClassFileContent$aClass$1.compute(KotlinBinaryClassCache.kt:73)
	at com.intellij.mock.MockApplication.runReadAction(MockApplication.java:178)
	at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion.getKotlinBinaryClassOrClassFileContent(KotlinBinaryClassCache.kt:73)
	at org.jetbrains.kotlin.load.kotlin.KotlinBinaryClassCache$Companion.getKotlinBinaryClassOrClassFileContent$default(KotlinBinaryClassCache.kt:59)
	at org.jetbrains.kotlin.load.kotlin.VirtualFileFinder.findKotlinClassOrContent(VirtualFileFinder.kt:34)
	at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope$classes$1.invoke(LazyJavaPackageScope.kt:62)
	at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope$classes$1.invoke(LazyJavaPackageScope.kt:54)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
	at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope.findClassifier(LazyJavaPackageScope.kt:142)
	at org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaPackageScope.getContributedClassifier(LazyJavaPackageScope.kt:132)
	at org.jetbrains.kotlin.load.java.lazy.descriptors.JvmPackageScope.getContributedClassifier(JvmPackageScope.kt:55)
	at org.jetbrains.kotlin.resolve.scopes.ChainedMemberScope.getContributedClassifier(ChainedMemberScope.kt:35)
	at org.jetbrains.kotlin.resolve.scopes.AbstractScopeAdapter.getContributedClassifier(AbstractScopeAdapter.kt:44)
	at org.jetbrains.kotlin.resolve.LazyExplicitImportScope.getContributedClassifier(LazyExplicitImportScope.kt:43)
	at org.jetbrains.kotlin.resolve.lazy.LazyImportScope$getClassifier$1.invoke(LazyImportScope.kt:256)
	at org.jetbrains.kotlin.resolve.lazy.LazyImportScope$getClassifier$1.invoke(LazyImportScope.kt:251)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager.compute(LockBasedStorageManager.java:290)
	at org.jetbrains.kotlin.resolve.lazy.LazyImportScope.getClassifier(LazyImportScope.kt:251)
	at org.jetbrains.kotlin.resolve.lazy.LazyImportScope.getContributedClassifier(LazyImportScope.kt:247)
	at org.jetbrains.kotlin.resolve.scopes.utils.ScopeUtilsKt.findClassifier(ScopeUtils.kt:87)
	at org.jetbrains.kotlin.resolve.QualifiedExpressionResolver.resolveToPackageOrClassPrefix(QualifiedExpressionResolver.kt:516)
	at org.jetbrains.kotlin.resolve.QualifiedExpressionResolver.resolveToPackageOrClassPrefix$default(QualifiedExpressionResolver.kt:480)
	at org.jetbrains.kotlin.resolve.QualifiedExpressionResolver.resolveToPackageOrClass(QualifiedExpressionResolver.kt:468)
	at org.jetbrains.kotlin.resolve.QualifiedExpressionResolver.resolveQualifierPartListForType(QualifiedExpressionResolver.kt:141)
	at org.jetbrains.kotlin.resolve.QualifiedExpressionResolver.resolveDescriptorForType(QualifiedExpressionResolver.kt:129)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveDescriptorForType(TypeResolver.kt:931)
	at org.jetbrains.kotlin.resolve.TypeResolver$resolveTypeElement$1.visitUserType(TypeResolver.kt:246)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitUserType(KtVisitorVoid.java:929)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitUserType(KtVisitorVoid.java:21)
	at org.jetbrains.kotlin.psi.KtUserType.accept(KtUserType.java:42)
	at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:60)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveTypeElement(TypeResolver.kt:244)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolvePossiblyBareType(TypeResolver.kt:123)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveType(TypeResolver.kt:113)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveTypeProjections(TypeResolver.kt:884)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveTypeForClass(TypeResolver.kt:504)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveTypeForClassifier(TypeResolver.kt:471)
	at org.jetbrains.kotlin.resolve.TypeResolver$resolveTypeElement$1.visitUserType(TypeResolver.kt:262)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitUserType(KtVisitorVoid.java:929)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitUserType(KtVisitorVoid.java:21)
	at org.jetbrains.kotlin.psi.KtUserType.accept(KtUserType.java:42)
	at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:60)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveTypeElement(TypeResolver.kt:244)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolvePossiblyBareType(TypeResolver.kt:123)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveType(TypeResolver.kt:113)
	at org.jetbrains.kotlin.resolve.TypeResolver.resolveType(TypeResolver.kt:84)
	at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.resolveValueParameters(FunctionDescriptorResolver.kt:421)
	at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.createValueParameterDescriptors(FunctionDescriptorResolver.kt:298)
	at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.initializeFunctionDescriptorAndExplicitReturnType(FunctionDescriptorResolver.kt:191)
	at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.resolveFunctionDescriptor(FunctionDescriptorResolver.kt:124)
	at org.jetbrains.kotlin.resolve.FunctionDescriptorResolver.resolveFunctionDescriptor(FunctionDescriptorResolver.kt:91)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.getDeclaredFunctions(AbstractLazyMemberScope.kt:115)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.access$getDeclaredFunctions(AbstractLazyMemberScope.kt:36)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$declaredFunctionDescriptors$1.invoke(AbstractLazyMemberScope.kt:56)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$declaredFunctionDescriptors$1.invoke(AbstractLazyMemberScope.kt:56)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunctionToNotNull.invoke(LockBasedStorageManager.java:651)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.doGetFunctions(AbstractLazyMemberScope.kt:96)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.access$doGetFunctions(AbstractLazyMemberScope.kt:36)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$functionDescriptors$1.invoke(AbstractLazyMemberScope.kt:49)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope$functionDescriptors$1.invoke(AbstractLazyMemberScope.kt:49)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager$MapBasedMemoizedFunctionToNotNull.invoke(LockBasedStorageManager.java:651)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.getContributedFunctions(AbstractLazyMemberScope.kt:92)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.AbstractLazyMemberScope.computeDescriptorsFromDeclaredElements(AbstractLazyMemberScope.kt:227)
	at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyPackageMemberScope.getContributedDescriptors(LazyPackageMemberScope.kt:44)
	at org.jetbrains.kotlin.resolve.scopes.ResolutionScope$DefaultImpls.getContributedDescriptors$default(ResolutionScope.kt:50)
	at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor.getDescriptorsWithKind(DefaultDescriptorToDocumentableTranslator.kt:687)
	at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor$visitPackageFragmentDescriptor$2.invokeSuspend(DefaultDescriptorToDocumentableTranslator.kt:131)
	at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor$visitPackageFragmentDescriptor$2.invoke(DefaultDescriptorToDocumentableTranslator.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:194)
	at org.jetbrains.dokka.base.translators.descriptors.DokkaDescriptorVisitor.visitPackageFragmentDescriptor(DefaultDescriptorToDocumentableTranslator.kt:130)
	at org.jetbrains.dokka.base.translators.descriptors.DefaultDescriptorToDocumentableTranslator$$special$$inlined$parallelMap$1$1.invokeSuspend(parallelCollectionOperations.kt:19)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

**Comment by michaldrabik** [comment]
Found any solutions @nate-eisner ?

**Comment by liedQM** [comment]
I see the same error like @nate-eisner with Java 17 installed.

**Comment by jisungbin** [comment]
If you have this problem during maven publishing, you can temporarily disable this task (this works great!).

```kotlin
// https://github.com/Kotlin/dokka/issues/2956
tasks.matching { task ->
  task.name.contains("javaDocReleaseGeneration", ignoreCase = true)
}.configureEach {
  enabled = false
}
```

**Comment by matejdro** [comment]
Wouldn't then that not generate the javadoc (whole point of running dokka)?

**Comment by stevenzeck** [comment]
We're running into this issue too, but I'm not convinced it's purely a Dokka issue. The step that fails during execution, `com.android.build.gradle.tasks.JavaDocGenerationTask$DokkaWorkAction`, looks to come from [AGP](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/tasks/JavaDocGenerationTask.kt). And it still errors after removing Dokka related plugins, Gradle tasks, etc.

I've tried creating a minimal repo that recreates this, but I haven't had any luck yet. All of the files that it shows in the errors. are sealed classes, but not all sealed classes are erroring.

**Comment by lukaszkalnik** [comment]
Running into the same problem.
Kotlin 1.9.22
AGP 8.3.0
Dokka 1.9.10
Lottie Compose 6.3.0

```
ERROR: Exception while analyzing expression at (238,45) in /runner/work/app-path-redacted/AppComposeScreen.kt
...
Caused by: java.lang.IllegalStateException: Could not read class: VirtualFile: /home/automata/.gradle/caches/transforms-4/ec27e9b5bc98114d7404e1026e605152/transformed/lottie-compose-6.3.0-api.jar!/com/airbnb/lottie/compose/LottieCompositionSpec.class
...
Caused by: java.lang.UnsupportedOperationException: PermittedSubclasses requires ASM9
```

**Comment by lukaszkalnik** [comment]
According to this [Slack message](https://kotlinlang.slack.com/archives/C0F4UNJET/p1709636032265459?thread_ts=1709575234.851329&cid=C0F4UNJET), at least my problem has been caused by AGP bundling some older version of Dokka, which activates when in the AGP's `publishing` DSL you use `withJavadocJar()`.
```
android {
    publishing {
        singleVariant("release") {
            withJavadocJar()
        }
    }
}
```
The workaround was to remove `withSourcesJar()` from the AGP DSL and instead include the generated Dokka output in top-level Gradle-DSL `artifacts` block:
```
android {
    publishing {
        singleVariant("release") {}
    }
}

tasks.dokkaHtml.configure {
    dokkaSourceSets {
        named("main") {
            noAndroidSdkLink.set(false)
        }
    }
}

tasks.register<Jar>("javadocJar") {
    archiveClassifier.set("javadoc")
    from("$buildDir/dokka/javadoc")
}

artifacts {
    tasks.findByName("javadocJar")
}
```

**Comment by philips77** [comment]
Hello,
I found 2 ways to solve the issue:
1. Decrease `jvmTarget`, `sourceCompatibility` and `targetCompatibility` from Java 17 to Java 11
2. Implement Dokka like in the previous comment.

As we had Kotlin plugins configured, here's the commit implementing the same in Kotlin:
https://github.com/NordicSemiconductor/Android-Gradle-Plugins/commit/e8a098049df76ca35b5d6a3d3ef92e960c9f84fc

**Comment by tprochazka** [comment]
We have the same issue after changing targetJVM to 17 from 11.
I created a thread for that also here:
https://kotlinlang.slack.com/archives/C0F4UNJET/p1721138854429639
I tried even a master build of dokka and it has the same issue. 
It is really annoying.

**Comment by yschimke** [comment]
Bumping to AGP 8.6.0-rc01 seems to have fixed this for me.

**Comment by tprochazka** [comment]
You can basically stop developing Dokka, it is useless with this bug, we completely stopped using it because of that. 
Related thread here
https://slack-chats.kotlinlang.org/t/22711948/i-have-one-really-strange-issue-with-dokka-java-lang-unsuppo

**Comment by whyoleg** [comment]
I want to reiterate: there is nothing we can do on the Dokka side, as the issue is on the AGP side.

See https://issuetracker.google.com/issues/460481088 for future updates on that matter.
It states that the recently bundled Dokka was updated in AGP, so the issue will be resolved in a future release (AGP 9.3-alpha05)

In the meantime, the only solution is not to use the Dokka bundled with AGP, as shown in the comment above (https://github.com/Kotlin/dokka/issues/2956#issuecomment-1978964475).
